src/tox/session/commands/show_config.py

```diff
@@ -57,17 +60,21 @@ def tox_info(config, parser):
 
 
 def version_info(parser):
-    import pkg_resources
-
     versions = OrderedDict()
-    visited = set()
     to_visit = {"tox"}
     while to_visit:
         current = to_visit.pop()
-        visited.add(current)
-        current_dist = pkg_resources.get_distribution(current)
-        to_visit.update(i.name for i in current_dist.requires() if i.name not in visited)
-        versions[current] = current_dist.version
+        current_dist = importlib_metadata.distribution(current)
+        current_name = canonicalize_name(current_dist.metadata["name"])
+        versions[current_name] = current_dist.version
+        if current_dist.requires is not None:
+            for require in current_dist.requires:
+                pkg = Requirement(require)
+                pkg_name = canonicalize_name(pkg.name)
+                if (
+                    pkg.marker is None or pkg.marker.evaluate({"extra": ""})
```

Comment:
can you explain this check here 🤔

Reply:
yeah this is a little bit weird

it's essentially a reimplement of `pkg_resources.Requirement.requires` but using the importlib-metadata / packaging primitives

the pkg_resources api has:

```python
    def requires(self, extras=()):
```

so by default it does not pass extras at all.

the `importlib_metadata` requires ends up looking like this for tox:

```pycon
>>> for req in importlib_metadata.distribution('tox').requires:
...     print(f'- {req}')
... 
- setuptools>=30.0.0
- pluggy<1,>=0.3.0
- py<2,>=1.4.17
- six<2,>=1.0.0
- virtualenv>=14.0.0
- toml>=0.9.4
- filelock<4,>=3.0.0
- sphinx<3,>=2.0.0; extra == "docs"
- towncrier>=18.5.0; extra == "docs"
- pygments-github-lexers>=0.0.5; extra == "docs"
- sphinxcontrib-autoprogram>=0.1.5; extra == "docs"
- freezegun<1,>=0.3.11; extra == "testing"
- pathlib2<3,>=2.3.3; extra == "testing"
- pytest<5,>=3.0.0; extra == "testing"
- pytest-cov<3,>=2.5.1; extra == "testing"
- pytest-mock<2,>=1.10.0; extra == "testing"
- pytest-xdist<2,>=1.22.2; extra == "testing"
- pytest-randomly<2,>=1.2.3; extra == "testing"
- flaky<4,>=3.4.0; extra == "testing"
- psutil<6,>=5.6.1; (python_version != "3.4") and extra == "testing"
```

Converting one of those later ones into `packaging` primitives:

```pycon
>>> from packaging.requirements import Requirement
>>> req = Requirement('pytest-randomly<2,>=1.2.3; extra == "testing"')
>>> req.marker
<Marker('extra == "testing"')>
>>> req.marker.evaluate()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/tox/venv/lib/python3.6/site-packages/packaging/markers.py", line 296, in evaluate
    return _evaluate_markers(self._markers, current_environment)
  File "/tmp/tox/venv/lib/python3.6/site-packages/packaging/markers.py", line 221, in _evaluate_markers
    lhs_value = _get_env(environment, lhs.value)
  File "/tmp/tox/venv/lib/python3.6/site-packages/packaging/markers.py", line 203, in _get_env
    "{0!r} does not exist in evaluation environment.".format(name)
packaging.markers.UndefinedEnvironmentName: 'extra' does not exist in evaluation environment.
>>> req.marker.evaluate({'extra': ''})
False
>>> req.marker.evaluate({'extra': 'testing'})
True
```